php-checks/src/main/java/org/sonar/php/checks/EncryptionModeAndPaddingCheck.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.FunctionArgumentCheck;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+
+@Rule(key = "S5542")
+public class EncryptionModeAndPaddingCheck extends FunctionArgumentCheck {
+
+  private static final String MESSAGE = "Use secure mode and padding scheme.";
+
+  private static final String OPENSSL_PUBLIC_ENCRYPT = "openssl_public_encrypt";
+  private static final String OPENSSL_PUBLIC_ENCRYPT_COMPLIANT_VALUE = "OPENSSL_PKCS1_OAEP_PADDING";
+
+  private static final String OPENSSL_ENCRYPT = "openssl_encrypt";
+  private static final Set<String> OPENSSL_PUBLIC_ENCRYPT_NONCOMPLIANT_VALUES = opensslPublicEncryptNoncompliantValues();
```

Comment:
I think we should remove "public" from the name of the constant and from the name of the method: we use them to check calls to `openssl_encrypt` (and not to `openssl_public_encrypt`)